src/main/java/teammates/ui/webapi/CreateReadNotificationAction.java

```diff
@@ -0,0 +1,54 @@
+package teammates.ui.webapi;
+
+import java.time.Instant;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.AccountAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Logger;
+import teammates.ui.output.AccountData;
+import teammates.ui.request.InvalidHttpRequestBodyException;
+import teammates.ui.request.ReadNotificationCreateRequest;
+
+/**
+ * Action: Creates a read notification in account entity.
+ */
+public class CreateReadNotificationAction extends Action {
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    void checkSpecificAccessControl() throws UnauthorizedAccessException {
+        // Any user can create a read status for notification.
+    }
+
+    @Override
+    public ActionResult execute() throws InvalidHttpRequestBodyException, InvalidOperationException {
+        ReadNotificationCreateRequest readNotificationCreateRequest =
+                getAndValidateRequestBody(ReadNotificationCreateRequest.class);
+        String notificationId = readNotificationCreateRequest.getNotificationId();
+        Instant endTime = Instant.ofEpochMilli(readNotificationCreateRequest.getEndTimestamp());
+
+        try {
+            if (!logic.doesNotificationExists(notificationId)) {
+                throw new InvalidParametersException("Invalid notification id.");
+            }
```

Comment:
Is this necessary? I think this check was also done in `updateReadNotifications`.

Reply:
Thank you! I think it is a good point.